airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/avro_parser.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from typing import Iterator, TextIO, BinaryIO, Union, Mapping
+from fastavro import reader
+import fastavro
+
+from .abstract_file_parser import AbstractFileParser
+from typing import Any
+
+
+# mapping from apache avro docs: https://avro.apache.org/docs/current/spec.html#schema_complex
+data_type_mapping = {
+    "null": "null",
+    "boolean": "boolean",
+    "int": "integer",
+    "long": "integer",
+    "float": "number",
+    "double": "number",
+    "bytes": "string",
+    "string": "string",
+    "record": "object",
+    "enum": "string",
+    "array": "array",
+    "map": "object",
+    "fixed": "string",
+}
+
+
+class AvroParser(AbstractFileParser):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def is_binary(self) -> bool:
+        return True
+
+    @staticmethod
+    def parse_data_type(data_type_mapping: dict, avro_schema: dict) -> dict:
```

Comment:
any specific reason this is a `@staticmethod`?

Also, for correctness we should prefix this function with `_` since it's only used privately within this class

Reply:
Removed staticmethod and added underscore